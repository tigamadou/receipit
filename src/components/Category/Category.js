import React from 'react';
import PropTypes from 'prop-types';
import {
  Link,
} from 'react-router-dom';

const Category = ({ category }) => (
  <div>
    <div>
      <img src={category.strCategoryThumb} alt={category.strCategory} />
      {category.strCategory}
    </div>
    <div>
      <Link to={`/category/${category.strCategory}`}> Details </Link>
    </div>
  </div>
);

Category.defaultProps = {
  category: null,
};

Category.propTypes = {
  category: PropTypes.instanceOf(Object),
};
export default Category;
