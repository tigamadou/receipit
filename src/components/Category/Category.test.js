import React from 'react';
import { render, screen } from '@testing-library/react';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import renderer from 'react-test-renderer';
import Category from './Category';

describe('Category component tests', () => {
  const data = {
    strCategoryThumb: 'thumb',
    strCategory: 'testCategory',

  };
  test('renders Category component ', () => {
    render(<Router><Category category={data} /></Router>);
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText('testCategory')).toBeInTheDocument;
  });

  test('should match with snapshot', () => {
    const tree = renderer
      .create(<Router><Category category={data} /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
