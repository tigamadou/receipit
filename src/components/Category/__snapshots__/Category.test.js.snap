// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component tests should match with snapshot 1`] = `
<div
  className="react-reveal card withOverlay"
  style={
    Object {
      "backgroundImage": "url(thumb)",
      "opacity": undefined,
    }
  }
>
  <a
    href="/category/testCategory"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="overlay"
      >
        <div>
          <h2
            className="title white is_xl"
          >
            testCategory
          </h2>
        </div>
        <div
          className="description"
        >
          <p
            className="is_sm"
          >
            Find all our
            testCategory
             
            recipes.
          </p>
        </div>
      </div>
    </div>
  </a>
</div>
`;
